src/base_contract.js

```diff
@@ -0,0 +1,133 @@
+import _ from 'lodash';
+import { Contract } from 'qweb3';
+
+import Config from '../config/config';
+import ContractMetadata from '../config/contract_metadata';
+
+const BaseContract = {
+  version: async function(args) {
+    const {
+      contractAddress, // address
+      senderAddress, // address
+    } = args;
+
+    if (_.isUndefined(contractAddress)) {
```

Comment:
I see these two defined checks used a lot. If you like you could use paramsCheck func defined in utils. It checks existence of necessary params in arg and throws Error if missing. You can modify it to throw TypeError.

In this case
paramsCheck("version", args, ["contractAddress","senderAddress"]);

The first arg methodName is only used for output.

Reply:
I'll think about refactoring all of this in another PR.